lib/liquid/condition.rb

```diff
@@ -47,6 +48,11 @@ def evaluate(context = Context.new)
       when :or
         result || @child_condition.evaluate(context)
       when :and
+        @@depth += 1
```

Comment:
Don't store this in global state.  Instead, use the context to keep state local to this liquid rendering